config/config.go

```diff
@@ -12,17 +18,170 @@ type contextValue struct {
 
 // Global configuration
 type Config struct {
-	Transitive             bool
-	TransitiveDepth        int
-	IncludeDevDependencies bool
-	Paranoid               bool
+	Transitive             bool `mapstructure:"transitive"`
+	TransitiveDepth        int  `mapstructure:"transitive_depth"`
+	IncludeDevDependencies bool `mapstructure:"include_dev_dependencies"`
+	Paranoid               bool `mapstructure:"paranoid"`
 
 	// DryRun to check for packages for risks.
 	// Do not actually execute any commands.
-	DryRun bool
+	DryRun bool `mapstructure:"dry_run"`
 
 	// InsecureInstallation allows bypassing install blocking on malicious packages
-	InsecureInstallation bool
+	InsecureInstallation bool `mapstructure:"insecure_installation"`
+
+	// TrustedPackages allows for trusting an suspicious package and ignoring the suspicious behaviour for the package in future installations
+	TrustedPackages map[string][]string `mapstructure:"trusted_packages"`
+}
+
+var (
+	setupOnce sync.Once
+	setupErr  error
+)
+
+// ErrConfigAlreadyExists is returned when creating the config without force and it already exists.
+var ErrConfigAlreadyExists = errors.New("pmg config already exists")
+
+// DefaultConfig returns the canonical default configuration used by PMG.
+func DefaultConfig() Config {
+	return Config{
+		Transitive:             true,
+		TransitiveDepth:        5,
+		IncludeDevDependencies: false,
+		Paranoid:               false,
+		DryRun:                 false,
+		InsecureInstallation:   false,
+		TrustedPackages:        map[string][]string{},
+	}
+}
+
+func configAsMap(cfg Config) map[string]any {
+	return map[string]any{
+		"transitive":               cfg.Transitive,
+		"transitive_depth":         cfg.TransitiveDepth,
+		"include_dev_dependencies": cfg.IncludeDevDependencies,
+		"dry_run":                  cfg.DryRun,
+		"paranoid":                 cfg.Paranoid,
+		"insecure_installation":    cfg.InsecureInstallation,
+		"trusted_packages":         cfg.TrustedPackages,
+	}
+}
+
+func SetupViper() (string, error) {
```

Comment:
`Viper` is internal to config package. Should not be exposed.

The config package should have simple method like `config.Load()` which loads the config from file if available, else returns default config.

We can have `config.Setup` which checks if config file does not exists and only then writes the default config to file.

We should keep minimum functions (contract) exposed from the config package. The rest of PMG should just use `Config` struct returned by `config.Load()` during its runtime.